### PR TITLE
Add !mute & !silent

### PR DESCRIPTION
**Pull request by PoorlyDefinedBehaviour**
Add two comands: !mute & !silent
Add `arg` to context object, which can be used instead of `args[0]` when the command takes only one argument

Example:

- Before
```ts
public async run({args}: Context): Promise<void> {
  foo(args[0])
  ...
}
```
- Now
```ts
public async run({arg}: Context): Promise<void> {
  foo(arg)
  ...
}
```

`args` can still be used when the command takes more than one argument(more than one word separated by spaces)

### CODE REVIEW

File: src/Core/Transformers/Message.ts [hunk 1]
```diff
@@ -53,7 +54,9 @@ export default class MessageTransformer {
       deleteChannelMessages: (options?: ChannelLogsQueryOptions) =>
         message.channel
           .fetchMessages(options)
-          .then((messages) => message.channel.bulkDelete(messages))
+          .then((messages) => message.channel.bulkDelete(messages)),
+      getMentionedUsers: () => message.mentions.members,
+      hasMentionedUsers: () => !!message.mentions.members.first()
```
Comment (EduardoRFS) [inline src/Core/Transformers/Message.ts]: This is just a suggestion so not blocking
```suggestion
      hasMentionedUsers: () => Boolean(message.mentions.members.first())
```

File: src/Core/Transformers/Message.ts [hunk 2]
```diff
@@ -24,8 +24,9 @@ export default class MessageTransformer {
       client,
       message,
       command,
+      arg: args.length > 0 ? args[0] : "",
       args,
-      members: client.guilds.get(process.env.GUILD_ID!)!.members,
+      members: {} as any /* client.guilds.get(process.env.GUILD_ID!)!.members */,
```
Comment (EduardoRFS) [inline src/Core/Transformers/Message.ts]: ## Blocking

Why? If this field isn't going to be used anymore, you should remove it and fix all the conflicts

File: src/Core/Transformers/Message.ts [hunk 3]
```diff
@@ -24,8 +24,9 @@ export default class MessageTransformer {
       client,
       message,
       command,
+      arg: args.length > 0 ? args[0] : "",
```
Comment (EduardoRFS) [inline src/Core/Transformers/Message.ts]:
```suggestion
      // TODO: we should discuss about it being defaulted to ""
      arg: args[0] || ""
```

File: src/Commands/Reuniao.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    return ":x: Como usar: `!reuniao <on|off>`"
+  }
+
+  public validate({ arg }: Context): void | never {
+    const states = {
+      on: 1,
+      off: 2
+    }
+
+    if (!(arg in states)) {
```
Comment (EduardoRFS) [inline src/Commands/Reuniao.ts]:
```suggestion
    // TODO: using in here it's a weird choice
    if (!(arg in states)) {
```

File: src/Commands/Reuniao.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+      throw new InvalidArgsException(this.help())
+    }
+  }
+
+  public async run({ client, arg }: Context): Promise<void> {
+    client.channels
+      .filter(
+        (channel) =>
+          !/Reunião/i.test((channel as any).name) &&
+          !/My Bot Server/i.test((channel as any).name)
```
Comment (EduardoRFS) [inline src/Commands/Reuniao.ts]: as @kjkGustavo pointed out at live, I think that using the ids here is a better idea, also it should be placed on a settings file and not be hard coded, I'm sure that we will delete the channel by accident sometime in the future.

File: src/Commands/Mute.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+      .addField("``📣`` **Motivo:**", muteReason, true)
+      .setThumbnail(userToMute.user.avatarURL)
+      .setColor("#8146DC")
+      .setFooter(
+        "2019 © He4rt Developers",
+        "https://heartdevs.com/wp-content/uploads/2018/12/logo.png"
+      )
+      .setTimestamp()
+
+    await send(
```
Comment (EduardoRFS) [inline src/Commands/Mute.ts]:
```suggestion
    // TODO: we should look if it's possible to send message in a batch preserving the order
    await send(
```

File: src/Commands/Reuniao.ts (hunk 4 above)
Comment (PoorlyDefinedBehaviour) [reply src/Commands/Reuniao.ts]: @kjkGustavo is working on this command at the moment, so i'll leave it as it is for now

**Pull request merged**